packages/wallet/src/services/TransactionsTracker.ts

```diff
@@ -127,6 +125,7 @@ export const createAddressTransactionsProvider = ({
         )
       ),
       combineLatest([addresses$, storedTransactions$.pipe(defaultIfEmpty([] as Cardano.HydratedTx[]))]).pipe(
+        // eslint-disable-next-line sonarjs/cognitive-complexity
```

Comment:
Can we hoist some logic to separate functions?

Reply:
Done here: [d243142](https://github.com/input-output-hk/cardano-js-sdk/pull/1530/commits/d243142eda345212e22f501f14dac1dce2f9df30)